pkg/resourceutil/sort_test.go

```diff
@@ -15,33 +15,99 @@
-	}
-
 	t.Run("Reordering resources based on default reordering function", func(t *testing.T) {
-		expected := []string{"PodSecurityPolicy", "ServiceAccount", "ClusterRole", "ClusterRoleBinding", "Secret", "ConfigMap", "ConfigMap", "Service", "Deployment", "IngressRoute", "Unknown"}
+		resources := []Resource{
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Unknown"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Secret"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ConfigMap"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ClusterRole"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "IngressRoute"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ClusterRoleBinding"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ConfigMap"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Deployment"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "PodSecurityPolicy"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ServiceAccount"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Service"}},
+		}
+		expected := []string{
+			"PodSecurityPolicy",
+			"ServiceAccount",
+			"ClusterRole",
+			"ClusterRoleBinding",
+			"Secret",
+			"ConfigMap",
+			"ConfigMap",
+			"Service",
+			"Deployment",
+			"IngressRoute",
+			"Unknown",
+		}
+		var orderedNames []string
+		for _, resource := range SortResourcesByKind(resources, nil) {
+			orderedNames = append(orderedNames, resource.GroupVersionKind.Kind)
+		}
+
+		assertDeepEqual(t, orderedNames, expected)
+	})
+
+	t.Run("Reordering resources with mia-platform.eu/apply-before-kinds annotation", func(t *testing.T) {
+		resources := []Resource{
+			makeResourceWithAnnotation(t, "UnknownBeforeNamespace", "Namespace"),
+			makeResourceWithAnnotation(t, "UnknownBeforeEmpty", ""),
+			makeResourceWithAnnotation(t, "UnknownBeforeDeploymentAndJob", "Job, Deployment"),
+			makeResourceWithAnnotation(t, "UnknownBeforeDeploymentAndUnknown", "Deployment, Unknown"),
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Job"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Pod"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Deployment"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Namespace"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Unknown"}},
+		}
+
+		expected := []string{
+			"UnknownBeforeNamespace",
+			"Namespace",
+			"Pod",
+			"UnknownBeforeDeploymentAndJob",
+			"UnknownBeforeDeploymentAndUnknown",
+			"Deployment",
+			"Job",
+			"UnknownBeforeEmpty",
+			"Unknown",
+		}
 		var orderedNames []string
 		for _, resource := range SortResourcesByKind(resources, nil) {
 			orderedNames = append(orderedNames, resource.GroupVersionKind.Kind)
 		}
-		require.ElementsMatch(t, expected, orderedNames)
+
+		assertDeepEqual(t, orderedNames, expected)
+	})
+}
+
+func assertDeepEqual(t *testing.T, actual, expected interface{}) {
```

Comment:
Why do not use the already imported https://github.com/stretchr/testify?

Reply:
changed with `assert.Exactly`